code/obj/item.dm

```diff
@@ -538,6 +552,10 @@
 /obj/item/temperature_expose(datum/gas_mixture/air, temperature, volume)
 	if (src.burn_possible && !src.burning)
 		if ((temperature > T0C + src.burn_point) && prob(5))
+			for (var/obj/item/I in get_turf(src))
+				if (I.firesource && !isnull(I))
+					for (var/obj/item/A in get_turf(src))
+						A.hotbox_item = I
```

Comment:
see comments about saving print instead. Also break the outer loop, no point in continuing onwards once you found the fire source.